### PR TITLE
Add color argument struct for cli applications

### PR DESCRIPTION
**Pull request by brendanzab**

### CODE REVIEW

File: codespan-reporting/examples/emit.rs [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
+use structopt::StructOpt;
 
 use codespan::{CodeMap, Span};
-use codespan_reporting::termcolor::{ColorChoice, StandardStream};
-use codespan_reporting::{emit, Diagnostic, Label, Severity};
+use codespan_reporting::termcolor::StandardStream;
+use codespan_reporting::{emit, ColorArg, Diagnostic, Label, Severity};
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "pikelet")]
```
[the remark this answers is not in the document]
Comment (brendanzab) [reply codespan-reporting/examples/emit.rs]: WOPS

[the remark this answers is not in the document]
Comment (brendanzab) [reply codespan-reporting/examples/emit.rs]: Yeah it was copy-paste 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️

File: codespan-reporting/src/lib.rs [hunk 2]
```diff
@@ -84,3 +85,67 @@ impl Severity {
...
+}
+
+impl FromStr for ColorArg {
+    type Err = &'static str;
+
+    fn from_str(src: &str) -> Result<ColorArg, &'static str> {
+        match src {
+            _ if src.eq_ignore_ascii_case("auto") => Ok(ColorArg(ColorChoice::Auto)),
+            _ if src.eq_ignore_ascii_case("always") => Ok(ColorArg(ColorChoice::Always)),
+            _ if src.eq_ignore_ascii_case("ansi") => Ok(ColorArg(ColorChoice::AlwaysAnsi)),
```
Comment (brendanzab) [inline codespan-reporting/src/lib.rs]: Do you think this should be `always-ansi`?

File: README.md [hunk 3]
```diff
@@ -14,7 +14,28 @@
...
-Utilities for dealing with source code locations and pretty diagnostic formatting
+Utilities for dealing with source code locations.
+
+## Supporting crates
+
+Codespan also allows you to easily set up pretty diagnostic formatting for
+command line interfaces via the [`codespan-reporting`][codespan-reporting]
+crate. This will give you output that looks like the following:
+
+![screenshot](./codespan-reporting/assets/screenshot.png)
```
[the remark this answers is not in the document]
Comment (brendanzab) [reply README.md]: You're right - excluded them!

File: codespan-reporting/src/lib.rs (hunk 2 above)
Comment (Marwes) [reply codespan-reporting/src/lib.rs]: I think it is fine to just have it as `ansi`, `AlwaysAnsi` in the enum seems rather redundant in the firstplace but perhaps it is better to be consistent and use `always-ansi`. Not sure.

Comment (Marwes) [reply codespan-reporting/src/lib.rs]: FWIW `rg` has just `ansi` on the command line.

Comment (brendanzab) [reply codespan-reporting/src/lib.rs]: Ah, might just go with `rg` then - I did notice you had `always-ansi` in Gluon. I wonder how much time has been wasted on this same question around the world! 😆

**Pull request merged**